packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3614,6 +3627,73 @@ export class TransactionController extends BaseController<
     return transactionMeta;
   }
 
+  async #updateFirstInteraction(
+    transactionMeta: TransactionMeta,
+    {
+      traceContext,
+    }: {
+      traceContext?: TraceContext;
+    } = {},
+  ) {
+    const {
+      chainId,
+      id: transactionId,
+      txParams: { to, from },
+    } = transactionMeta;
+
+    const request: FirstTimeInteractionRequest = {
+      chainId,
+      to,
+      from,
+    };
+
+    let firstTimeInteractionResponse: FirstTimeInteractionResponse;
+
+    try {
+      validateFirstTimeInteraction(request);
+
+      firstTimeInteractionResponse = await this.#trace(
+        { name: 'FirstTimeInteraction', parentContext: traceContext },
+        () => getFirstTimeInteraction(request),
```

Comment:
Do we need a toggle to disable this or is that a future ticket?

At what point is it easier to provide a single `isExternalApiDisabled` callback?

Reply:
Good call, I will clarify this with Barbara.